lib/dat-worker-pool/worker.rb

```diff
@@ -87,6 +87,10 @@ def raise(*args)
         @thread.raise(*args) if self.thread_alive?
       end
 
+      # Helpers - TODO make these private once callbacks are instance evald
+      def params; @runner.worker_params; end
```

Comment:
@kellyredding - We could mutex lock the params reading in the runner. That would make reading thread-safe but that doesn't help with changing the params. You would have to be responsible for making that thread-safe yourself, i.e.

```ruby
def work!
  params[:mutex].synchronize{ params[:value] = 'something' }
end
```

You definitely have to do something like this if you want to modify the params in the worker. There's not much we can do to help you, maybe provide a mutex for all workers by default and provide a helper method for using it. 

```ruby
def work!
  mutex_synchronize{ params[:value] = 'something' }
end
```

Part of using dat-worker-pool does require you to be thread-safe on your own but I don't mind doing anything we can to help.